### ISSUE TITLE
ArbMatrix(rand(BigFloat, 5,5))

### ISSUE THREAD

**Issue by kalmarek** [kalmarek/Arblib.jl#83]
fails due to missing `set!` method
```julia
ERROR: MethodError: no method matching set!(::Ptr{Arblib.arb_struct}, ::BigFloat)
Closest candidates are:
  set!(::ArbLike, ::Rational) at /home/kalmar/.julia/dev/Arblib/src/constructors.jl:153
  set!(::ArfLike, ::BigFloat) at /home/kalmar/.julia/dev/Arblib/src/arbcall.jl:294
  set!(::ArbLike, ::ArbLike) at /home/kalmar/.julia/dev/Arblib/src/arbcall.jl:294
  ...
Stacktrace:
 [1] setindex!(::Arblib.arb_mat_struct, ::BigFloat, ::Int64, ::Int64) at /home/kalmar/.julia/dev/Arblib/src/matrix.jl:41
 [2] setindex! at /home/kalmar/.julia/dev/Arblib/src/matrix.jl:49 [inlined]
 [3] ArbMatrix(::Array{BigFloat,2}; prec::Int64) at /home/kalmar/.julia/dev/Arblib/src/matrix.jl:114
 [4] ArbMatrix(::Array{BigFloat,2}) at /home/kalmar/.julia/dev/Arblib/src/matrix.jl:110
 [5] top-level scope at REPL[9]:1
```
(this is a reminder for #82)

**Comment by Joel-Dahne** [comment]
Ah, it's trying to use the `set!` method! My idea was to implement `set!` methods for all constructors we have. Most of the constructors are just wrappers of `set!` methods, but some of them call other methods instead. Creating `set!` methods for all of them would create a more unified interface. I could do this as part of the cleanup of constructors!

**Comment by kalmarek** [comment]
fixed (probably) by #76
